dimod/sampleset.py

```diff
@@ -997,6 +1002,90 @@ def aggregate(self):
 
         return type(self)(record, self.variables, self.info, self.vartype)
 
+    def append_variables(self, samples_like, sort_labels=True):
```

Comment:
`append_variables` could work. I still like `hstack`, the only problem is one could expect the semantics to extend to info vectors too. (If we went with `hstack`, I would ditch the special case of shape mismatch when adding just one sample.)

Reply:
Yes, ditching the single-sample case is what I did when I went to implement hstack. But since adding/multipling/joining are all valid options depending on the vector content, I think we will need to leave that to the user.

Reply:
for this case, adding just one sample is pretty important. that would be the default use case

Reply:
Yes, I think hstack is a different function with different logic. These two actions are complementary and trying to integrate them into a single function would just be confusing.

Reply:
Agreed. If we want to handle add-one-sample-to-many logic (we do), name `append_variables` is better.